core/src/main/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateAction.java

```diff
@@ -81,7 +81,8 @@ protected void masterOperation(final ClusterStateRequest request, final ClusterS
         if (request.routingTable()) {
             if (request.indices().length > 0) {
                 RoutingTable.Builder routingTableBuilder = RoutingTable.builder();
-                for (String filteredIndex : request.indices()) {
+                String[] indices = indexNameExpressionResolver.concreteIndexNames(currentState, request);
```

Comment:
interesting, seems like a related bug that we may need an issue for and new tests for?